src/clj/harja/palvelin/integraatiot/paikkatietojarjestelma/tuonnit/sillat.clj

```diff
@@ -192,20 +192,30 @@
 
 (defn vie-sillat-kantaan [db shapefile]
   (if shapefile
-    (let [siltatietueet-shapefilesta (shapefile/tuo shapefile)
+    (let [batch-koko 15  ;; Montako siltaa prosessoidaan yhdessä transaktiossa
+          siltatietueet-shapefilesta (shapefile/tuo shapefile)
           tallennettavat-siltatietueet (-> siltatietueet-shapefilesta
                                          suodata-sillat
                                          jarjesta-voimassaolevat-sillat-yksittaisille-riveille
                                          parsi-tieosoitteet)]
+      
       (log/debug (str "Tuodaan sillat kantaan tiedostosta " shapefile))
-      (try (jdbc/with-db-transaction [db db]
-             (doseq [silta tallennettavat-siltatietueet]
-               (vie-silta-entry db silta)))
-           (log/debug "siltojen tuonti kantaan valmis.")
-           (catch PSQLException e
-             (log/error "Siltojen tuonnissa kantaan tapahtui virhe: " e)
-             (throw e))
-           (catch Exception e
-             (log/error "Siltojen tuonnissa tapahtui virhe: " e)
-             (throw e))))
+      (log/debug (str "Siltojen määrä: " (count tallennettavat-siltatietueet)))
+
+      (dorun
```

Comment:
Dorun + for yhdistelmä on käytännössä doseq

Reply:
Joo, kävi kyllä mielessä, toi on oikeastaan vaan tota alempaa printtiä varten.  Voin siivota tätä.

En osaa nyt sanoo tarkalleen mitä tässä käy pellin alla, mun käsittääkseni **for** ainakin tekee loopista lazyn, joka parempi JVM muistin kannalta, mutta kyllä, dorunin sisällä ajo voi muuttua. Muistin käyttäytymisen perusteella kuitenkin, tää jostain syystä on silti parempi kun pelkkä doseq.

Mutta, siivoilen tota vähän, ja hyvä pointti, kiitos

Reply:
Doall kuulemma tarkoitettu lazyjen kanssa, joten wrappasin siihen. Muuten tämä koodi ei juokse. Testailin, ja muisti pysyy yhä kunnossa, lokaalisti ainakin.